From 8ddc72fcf132a260ddb3c3c07a1add29c759008b Mon Sep 17 00:00:00 2001
From: Khanh Le <khanh.le.xr@renesas.com>
Date: Mon, 14 Jul 2025 18:04:10 +0700
Subject: [PATCH 2/2] arm64: dts: renesas: gray-hawk: Add reserved area for
 Linux kernel image

After waking up from Low Power Mode (LPM), avoid reloading the Linux kernel
image by reusing the previously loaded image for a warm boot.
A reserved area is added to ensure the kernel image remains intact during
the first boot.

Signed-off-by: Khanh Le <khanh.le.xr@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a779h0-gray-hawk-common.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779h0-gray-hawk-common.dtsi b/arch/arm64/boot/dts/renesas/r8a779h0-gray-hawk-common.dtsi
index 45b4e715940c..b2b2a272ad37 100644
--- a/arch/arm64/boot/dts/renesas/r8a779h0-gray-hawk-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a779h0-gray-hawk-common.dtsi
@@ -124,6 +124,11 @@ ecc_data_region@5cccc0000 {
 			no-map;
 			reg = <0x5 0xcccc0000 0x0 0x33340000>;
 		};
+
+		kernel_image@48080000 {
+			no-map;
+			reg = <0x0 0x48080000 0x0 0x3c00000>;
+		};
 	};
 
 	sound_card: sound {
-- 
2.34.1

